dashboard/app/helpers/plc/evaluations_helper.rb

```diff
@@ -0,0 +1,34 @@
+module Plc::EvaluationsHelper
+  def determine_preferred_learning_modules level_source, script
+    return [] if level_source.nil?
+
+    responses = JSON.parse(level_source.data)
+    learning_module_weights = Hash.new(0)
+
+    responses.each do |level_id, response|
+      level = EvaluationMulti.find(level_id)
+      selected_answer = level.answers[response['result'].to_i]
+
+      next if selected_answer['stage'].nil?
+
+      stage = Stage.find_by(name: selected_answer['stage']['name'], script_id: script.id)
```

Comment:
this could probably be rewritten to use the script cache (you can experiment in `rails c`), something like `script.stages.select {|name| name == selected_answer['stage']['name']}`